apm-agent-core/src/test/java/co/elastic/apm/agent/logging/ApmServerLogAppenderTest.java

```diff
@@ -67,14 +67,24 @@ void createAndInit() throws Exception {
         appender.getInitListener().init(tracer);
 
         assertThatThrownBy(() -> appender.getInitListener().init(tracer), "should throw when trying to init more than once");
+
+        // testing singleton invariants, while being implementation details matter here
+        assertThat(ApmServerLogAppender.getInstance())
+            .describedAs("singleton instance should be set by the factory")
+            .isSameAs(appender);
+        assertThat(ApmServerLogAppender.createAppender("test", layout))
+            .describedAs("factory method should only create once and return the singleton afterwards")
+            .isSameAs(appender);
```

Comment:
[for reviewer] making sure I don't make the same mistake again, just in case I forget.